src/phpDocumentor/Transformer/Writer/Twig/EnvironmentFactory.php

```diff
@@ -44,29 +35,18 @@ public function create(
         Transformation $transformation,
         string $destination
     ) : Environment {
-        $callingTemplatePath = $this->getTemplatePath($transformation);
-
-        $baseTemplatesPath = $transformation->getTransformer()->getTemplates()->getTemplatesPath();
-
-        $templateFolders = [
-            $baseTemplatesPath . '/..' . DIRECTORY_SEPARATOR . $callingTemplatePath,
-            // http://twig.sensiolabs.org/doc/recipes.html#overriding-a-template-that-also-extends-itself
-            $baseTemplatesPath,
-        ];
-
-        // get all invoked template paths, they overrule the calling template path
-        /** @var Template $template */
-        foreach ($transformation->getTransformer()->getTemplates() as $template) {
-            $path = $baseTemplatesPath . DIRECTORY_SEPARATOR . $template->getName();
-            array_unshift($templateFolders, $path);
-        }
-
-        // Clone twig because otherwise we cannot re-set the extensions on this twig environment on every run of this
-        // writer
-        $env = new Environment(new FilesystemLoader($templateFolders));
-
-        $this->addPhpDocumentorExtension($project, $transformation, $destination, $env);
-        $this->addExtensionsFromTemplateConfiguration($transformation, $project, $env);
+        $mountManager = $transformation->template()->files();
+        $env = new Environment(
+            new ChainLoader(
```

Comment:
You are kind of doing the thing I suggested earlier because twig cannot handle multiple filesytems. Would have been nice if it was abstracted here... Maybe our own `FileSystemChain`?

Reply:
As I mentioned, not really. But it does make me wonder if this chain loader is actually incorrect. Although my tests with the previous iteration showed the same paths. A future investigation may be nice :)

And I prefer not to do this earlier because then we mix Twig specific functionality with filesystems; these two are not the same.

For example, you may want to copy specific paths to your destination and for that, it is best if you refer to a specific source filesystem. Otherwise, you may get similar issues to Requests using a generic 'get' where you are not sure whether the value comes from the query, post data or cookies.